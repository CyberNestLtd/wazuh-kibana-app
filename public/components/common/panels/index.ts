export * from './panel_split';
export { AggTable } from './agg-table';