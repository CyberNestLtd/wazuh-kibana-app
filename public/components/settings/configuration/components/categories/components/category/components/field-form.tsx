/*
 * Wazuh app - React component building the configuration component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React from 'react';
import {
  EuiCodeEditor,
  EuiImage,
  EuiFilePicker,
  EuiFieldNumber,
  EuiFieldText,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFormRow,
  EuiSelect,
  EuiSpacer,
  EuiSwitch,
  EuiTextArea,
} from '@elastic/eui';
import 'brace/mode/javascript';
import 'brace/snippets/javascript';
import 'brace/ext/language_tools';
import "brace/ext/searchbox";
import {
  UI_ERROR_SEVERITIES,
} from '../../../../../../../../react-services/error-orchestrator/types';
import { TPluginSettingWithKey, UI_LOGGER_LEVELS } from '../../../../../../../../../common/constants';
import { getErrorOrchestrator } from '../../../../../../../../react-services/common-services';
import _ from 'lodash';
import { WzButtonModalConfirm } from '../../../../../../../common/buttons';
import { WzRequest } from '../../../../../../../../react-services';
import { getAssetURL } from '../../../../../../../../utils/assets';
import { getHttp } from '../../../../../../../../kibana-services';
import { useDispatch } from 'react-redux';
import { updateAppConfig } from '../../../../../../../../redux/actions/appConfigActions';
import { useFormFieldChanged } from '../../../../../../../common/hooks';

interface IFieldForm {
  item: TPluginSettingWithKey
  value: any
  initialValue: any
  onChange: () => void
}

export const FieldForm: React.FunctionComponent<IFieldForm> = (props) => {
  const { item } = props;
  const ComponentForm = Form[item.type];

  if(ComponentForm){
    return <ComponentForm {...props}/>
  };

  return null;
};

//#region forms
const TextForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    initialValue,
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiFieldText
        fullWidth
        value={value}
        isInvalid={isInvalid}
        onChange={onChange} />
    </EuiFormRow>
  );
};

const TextAreaForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    initialValue,
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiTextArea
        fullWidth
        value={value}
        isInvalid={isInvalid}
        onChange={onChange} />
    </EuiFormRow>
  );
};

const NumberForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    initialValue,
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiFieldNumber
        fullWidth
        value={value}
        onChange={onChange} />
    </EuiFormRow>
  );
};

  
const BooleanForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    initialValue,
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiSwitch
        label={String(value)}
        checked={value}
        onChange={onChange} />
    </EuiFormRow>
  );
};

const ListForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    initialValue,
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiSelect
        options={item.options.choices}
        value={value}
        onChange={onChange} />
    </EuiFormRow>
  )
};

const IntervalForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    initialValue,
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiFieldText
        fullWidth
        value={value}
        isInvalid={isInvalid}
        onChange={onChange}
      />
    </EuiFormRow>
  );
}

const ArrayForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField}) => {
  const { value, error, onChange } = useFormFieldChanged(
    item.key,
    JSON.stringify(initialValue),
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <EuiCodeEditor
        mode='json'
        height='50px'
        width='100%'
        value={value}
        onChange={onChange}
    />
    </EuiFormRow>
  );
}

const FilePickerForm: React.FunctionComponent<IFieldForm> = ({item, initialValue, onChange: onChangeFormField, value: valueImage}) => {
  const dispatch = useDispatch();
  const filename = valueImage ? getHttp().basePath.prepend(getAssetURL(valueImage)) : null;

  const { error, onChange } = useFormFieldChanged(
    item.key,
    JSON.stringify(initialValue),
    {validate: item?.validate, onChange: onChangeFormField, type: item.type}
  );

  const isInvalid = Boolean(error);

  return (
    <EuiFormRow label={item.key} fullWidth isInvalid={isInvalid} error={error}>
      <>
        {filename && (
          <EuiFlexGroup alignItems="center">
            <EuiFlexItem grow={false}>
              <EuiImage src={filename} size='s'/>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <WzButtonModalConfirm
                buttonType="icon"
                tooltip={{
                  content: 'Delete file',
                  position: 'top',
                }}
                modalTitle={`Do you want to delete the ${item.key} file?`}
                onConfirm={async () => {
                  try{
                    await WzRequest.genericReq('DELETE', `/utils/configuration/files/${item.key}`);
                    dispatch(updateAppConfig({[item.key]: ''}));
                  }catch(error){
                    const options = {
                      context: `${FilePickerForm.name}.confirmDeletePolicy`,
                      level: UI_LOGGER_LEVELS.ERROR,
                      severity: UI_ERROR_SEVERITIES.BUSINESS,
                      store: true,
                      error: {
                        error: error,
                        message: error.message || error,
                        title: error.name || error,
                      },
                    };
                    getErrorOrchestrator().handleError(options);
                  }
                }}
                modalProps={{ buttonColor: 'danger' }}
                iconType="trash"
                color="danger"
                aria-label="Delete file"
              />
            </EuiFlexItem>
          </EuiFlexGroup>
        )}
        <EuiSpacer size='s' />
        <EuiFilePicker
          id={''}
          initialPromptText="Select or drag the file"
          onChange={(fileList) => onChange(fileList?.[0])}
          display={true ? 'large' : 'default'}
          fullWidth
          aria-label="Upload a file"
        />
      </>
    </EuiFormRow>
  );
}

const Form = {
  text: TextForm,
  number: NumberForm,
  boolean: BooleanForm,
  list: ListForm,
  array: ArrayForm,
  interval: IntervalForm,
  filepicker: FilePickerForm,
  textarea: TextAreaForm
}

//#endregion
