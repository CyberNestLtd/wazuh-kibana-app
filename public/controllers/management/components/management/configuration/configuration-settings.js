/*
 * Wazuh app - Definitions of configuration sections.
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { hasAgentSupportModule } from '../../../../../react-services/wz-agents';
import { WAZUH_MODULES_ID } from '../../../../../../common/constants';
import { i18n } from '@kbn/i18n';
const descp1 = i18n.translate('controller.manage.comp.confi.setting.descp1', {
  defaultMessage: 'Global and remote settings',
});
const descp2 = i18n.translate('controller.manage.comp.confi.setting.descp2', {
  defaultMessage: 'Master node configuration',
});
const descp3 = i18n.translate('controller.manage.comp.confi.setting.descp3', {
  defaultMessage: 'Automatic agent registration service',
});
const descp4 = i18n.translate('controller.manage.comp.confi.setting.descp4', {
  defaultMessage: 'Logging settings that apply to the agent',
});
const descp5 = i18n.translate('controller.manage.comp.confi.setting.descp5', {
  defaultMessage: 'Settings related to the connection with the manager',
});
const descp6 = i18n.translate('controller.manage.comp.confi.setting.descp6', {
  defaultMessage: 'Agent bucket parameters to avoid event flooding',
});
const descp7 = i18n.translate('controller.manage.comp.confi.setting.descp7', {
  defaultMessage: 'User-defined information about the agent included in alerts',
});
const descp8 = i18n.translate('controller.manage.comp.confi.setting.descp8', {
  defaultMessage: 'Settings related to the alerts and their format',
});
const descp9 = i18n.translate('controller.manage.comp.confi.setting.descp9', {
  defaultMessage:
    'Slack, VirusTotal and PagerDuty integrations with external APIs',
});
const descp10 = i18n.translate('controller.manage.comp.confi.setting.descp', {
  defaultMessage:
    'Configuration to ensure compliance with security policies, standards and hardening guides',
});
const descp11 = i18n.translate('controller.manage.comp.confi.setting.descp11', {
  defaultMessage:
    'Configuration assessment and automation of compliance monitoring using SCAP checks',
});
const descp12 = i18n.translate('controller.manage.comp.confi.setting.descp12', {
  defaultMessage: 'Configuration assessment using CIS scanner and SCAP checks',
});
const descp13 = i18n.translate('controller.manage.comp.confi.setting.descp13', {
  defaultMessage:
    'Discover what applications are affected by well-known vulnerabilities',
});
const descp14 = i18n.translate('controller.manage.comp.confi.setting.descp14', {
  defaultMessage:
    'Expose an operating system as a high-performance relational database',
});
const descp15 = i18n.translate('controller.manage.comp.confi.setting.descp15', {
  defaultMessage:
    'Gather relevant information about system OS, hardware, networking and packages',
});
const descp16 = i18n.translate('controller.manage.comp.confi.setting.descp16', {
  defaultMessage: 'Active threat addressing by immediate response',
});
const descp17 = i18n.translate('controller.manage.comp.confi.setting.descp17', {
  defaultMessage: 'Active threat addressing by immediate response',
});
const descp18 = i18n.translate('controller.manage.comp.confi.setting.descp18', {
  defaultMessage:
    'Monitor and collect the activity from Docker containers such as creation, running, starting, stopping or pausing events',
});
const descp19 = i18n.translate('controller.manage.comp.confi.setting.descp19', {
  defaultMessage:
    "Log analysis from text files, Windows events or syslog outputs'",
});
const descp20 = i18n.translate('controller.manage.comp.confi.setting.descp20', {
  defaultMessage:
    'Identify changes in content, permissions, ownership, and attributes of files',
});
const descp21 = i18n.translate('controller.manage.comp.confi.setting.descp21', {
  defaultMessage:
    'Run integrity checks on devices such as routers, firewalls and switches',
});
const descp22 = i18n.translate('controller.manage.comp.confi.setting.descp22', {
  defaultMessage:
    'Security events related to Amazon AWS services, collected directly via AWS API',
});
const descp23 = i18n.translate('controller.manage.comp.confi.setting.descp23', {
  defaultMessage: 'Configuration options of the Azure Logs wodle',
});
const descp24 = i18n.translate('controller.manage.comp.confi.setting.descp24', {
  defaultMessage: 'Configuration options of the Google Cloud Pub/Sub module',
});
const descp25 = i18n.translate('controller.manage.comp.confi.setting.descp25', {
  defaultMessage: 'Detect threats targeting GitHub organizations',
});
const descp26 = i18n.translate('controller.manage.comp.confi.setting.descp26', {
  defaultMessage: 'Configuration options of the Office 365 module',
});
const descp27 = i18n.translate('controller.manage.comp.confi.setting.descp27', {
  defaultMessage: 'Configuration options of the Command wodle',
});

const Title1 = i18n.translate('controller.manage.comp.confi.setting.Title1', {
  defaultMessage: 'Main configurations',
});
const Title2 = i18n.translate('controller.manage.comp.confi.setting.Title2', {
  defaultMessage: 'Alerts and output management',
});
const Title3 = i18n.translate('controller.manage.comp.confi.setting.Title3', {
  defaultMessage: 'Auditing and policy monitoring',
});
const Title4 = i18n.translate('controller.manage.comp.confi.setting.title4', {
  defaultMessage: 'System threats and incident response',
});
const Title5 = i18n.translate('controller.manage.comp.confi.setting.title5', {
  defaultMessage: 'Log data analysis',
});
const Title6 = i18n.translate('controller.manage.comp.confi.setting.title6', {
  defaultMessage: 'Cloud security monitoring',
});

export default [
  {
    title: Title1,
    description: '',
    settings: [
      {
        name: 'Global Configuration',
        description: descp1,
        goto: 'global-configuration',
        when: 'manager',
      },
      {
        name: 'Cluster',
        description: descp2,
        goto: 'cluster',
        when: 'manager',
      },
      {
        name: 'Registration Service',
        description: descp3,
        goto: 'registration-service',
        when: 'manager',
      },
      {
        name: 'Global Configuration',
        description: descp4,
        goto: 'global-configuration-agent',
        when: 'agent',
      },
      {
        name: 'Communication',
        description: descp5,
        goto: 'client',
        when: 'agent',
      },
      {
        name: 'Anti-flooding settings',
        description: descp6,
        goto: 'client-buffer',
        when: 'agent',
      },
      {
        name: 'Labels',
        description: descp7,
        goto: 'alerts-agent',
        when: 'agent',
      },
      // ,
      // { //TODO: Uncomment this to activate Log Settings
      //   name: 'Log settings',
      //   description: 'Alerts, archives and internal settings',
      //   goto: 'log-settings'
      // }
    ],
  },
  {
    title: Title2,
    description: '',
    settings: [
      {
        name: 'Alerts',
        description: descp8,
        goto: 'alerts',
        when: 'manager',
      },
      {
        name: 'Integrations',
        description: descp9,
        goto: 'integrations',
        when: 'manager',
      },
    ],
  },
  {
    title: Title3,
    description: '',
    settings: [
      {
        name: 'Policy monitoring',
        description: descp10,
        goto: 'policy-monitoring',
      },
      {
        name: 'OpenSCAP',
        description: descp11,
        goto: 'open-scap',
        when: agent => hasAgentSupportModule(agent, WAZUH_MODULES_ID.OPEN_SCAP),
      },
      {
        name: 'CIS-CAT',
        description: descp12,
        goto: 'cis-cat',
      },
    ],
  },
  {
    title: Title4,
    description: '',
    settings: [
      {
        name: 'Vulnerabilities',
        description: descp13,
        goto: 'vulnerabilities',
        when: 'manager',
      },
      {
        name: 'Osquery',
        description: descp14,
        goto: 'osquery',
      },
      {
        name: 'Inventory data',
        description: descp15,
        goto: 'inventory',
      },
      {
        name: 'Active Response',
        description: descp16,
        goto: 'active-response',
        when: 'manager',
      },
      {
        name: 'Active response',
        description: descp17,
        goto: 'active-response-agent',
        when: 'agent',
      },
      {
        name: 'Commands',
        description: descp27,
        goto: 'commands',
      },
      {
        name: 'Docker listener',
        description: descp18,
        goto: 'docker-listener',
        when: agent => hasAgentSupportModule(agent, WAZUH_MODULES_ID.DOCKER),
      },
    ],
  },
  {
    title: Title5,
    description: '',
    settings: [
      {
        name: 'Log collection',
        description: descp19,
        goto: 'log-collection',
      },
      {
        name: 'Integrity monitoring',
        description: descp20,
        goto: 'integrity-monitoring',
      },
      {
        name: 'Agentless',
        description: descp21,
        goto: 'agentless',
        when: 'manager',
      },
    ],
  },
  {
    title: Title6,
    description: '',
    settings: [
      {
        name: 'Amazon S3',
        description: descp22,
        goto: 'aws-s3',
      },
      {
        name: 'Azure Logs',
        description: descp23,
        goto: 'azure-logs',
        when: 'manager',
      },
      {
        name: 'Google Cloud Pub/Sub',
        description: descp24,
        goto: 'gcp-pubsub',
      },
      {
        name: 'GitHub',
        description: descp25,
        goto: 'github',
      },
      {
        name: 'Office 365',
        description: descp27,
        goto: 'office365',
        when: 'manager',
      },
    ],
  },
];
