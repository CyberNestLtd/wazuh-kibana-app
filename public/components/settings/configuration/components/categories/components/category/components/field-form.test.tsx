/*
 * Wazuh app - React test for FieldForm component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import React from 'react';
import { FieldForm } from './field-form';
import { ISetting } from '../../../../../configuration';
import { mount } from 'enzyme';
import { TPluginSettingWithKey } from '../../../../../../../../../common/constants';

describe('FieldForm component', () => {
  it('renders correctly to match the snapshot', () => {
    const item: TPluginSettingWithKey = {
      key: 'custom.setting',
      description: 'string',
      category: 1,
      name: 'Custom setting',
      type: 'text',
    };
    const updatedConfig = {};
    const setUpdatedConfig = jest.fn();

    const wrapper = mount(
      <FieldForm item={item} value={''} updatedConfig={updatedConfig} setUpdatedConfig={setUpdatedConfig} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
