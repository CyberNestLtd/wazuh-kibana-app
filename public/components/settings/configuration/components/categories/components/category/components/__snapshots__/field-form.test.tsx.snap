// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldForm component renders correctly to match the snapshot 1`] = `
<FieldForm
  item={
    Object {
      "category": 1,
      "description": "string",
      "key": "custom.setting",
      "name": "Custom setting",
      "type": "text",
    }
  }
  setUpdatedConfig={[MockFunction]}
  updatedConfig={Object {}}
  value=""
>
  <TextForm
    item={
      Object {
        "category": 1,
        "description": "string",
        "key": "custom.setting",
        "name": "Custom setting",
        "type": "text",
      }
    }
    setUpdatedConfig={[MockFunction]}
    updatedConfig={Object {}}
    value=""
  >
    <EuiFieldText
      fullWidth={true}
      onChange={[Function]}
      value=""
    >
      <EuiFormControlLayout
        fullWidth={true}
      >
        <div
          className="euiFormControlLayout euiFormControlLayout--fullWidth"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <EuiValidatableControl>
              <input
                className="euiFieldText euiFieldText--fullWidth"
                onChange={[Function]}
                type="text"
                value=""
              />
            </EuiValidatableControl>
            <EuiFormControlLayoutIcons />
          </div>
        </div>
      </EuiFormControlLayout>
    </EuiFieldText>
  </TextForm>
</FieldForm>
`;
